api/filters/nameref/nameref.go

```diff
@@ -86,6 +86,16 @@ func (f Filter) setScalar(node *yaml.RNode) error {
 	return nil
 }
 
+func filterReferralCandidates(referrer *resource.Resource, matches []*resource.Resource) []*resource.Resource {
+	ret := []*resource.Resource{}
```

Comment:
FWIW, this can be just
```
var ret []*resource.Resource
```
No need actually allocate the array.  `append` will do the right thing.